library/std/src/sys_common/thread_parker/generic.rs

```diff
@@ -16,13 +17,18 @@ pub struct Parker {
 }
 
 impl Parker {
-    pub fn new() -> Self {
-        Parker { state: AtomicUsize::new(EMPTY), lock: Mutex::new(()), cvar: Condvar::new() }
+    /// Construct the generic parker. The UNIX parker implementation
+    /// requires this to happen in-place.
+    pub unsafe fn new(parker: *mut Parker) {
```

Comment:
If this took a `&mut MaybeUninit<Parker>` instead of a `*mut Parker`, then this function could be safe.

Without MaybeUninit 'projections' though, that makes using this function tricky when the parker is part of a bigger MaybeUninit. :(

Reply:
That, and the lack of a dependency on `pin_project` in `std`, led me to use a pointer here, as we would require unsafe in `Thread::new` anyways. I could change this however.